### PR TITLE
Adding the Linter

### LINKED ISSUE #8: Add linter to project
## Main Goal
Add a linter to your project so that you don't accumulate hours looking for "the spot you forgot that last `)`"

### When Accomplished
- [x] Running `npm start` runs a linter on your javascript **before** serving files
- [x] The linters reports issues when found
  - [x] The reports are formatted decently

## Notes
- Linting is a weird way of saying "identify and report on patterns". Linters essentially make it easier to catch syntax mistakes and when breaking best practice. There are a lot out there, but I would suggest sticking with something our products at Banno use.

## Dependencies
- [x] #5

### LINKED ISSUE #24: Use Yarn over npm
## Main Goal
Have your repo "default" to using Yarn over npm

### When Accomplished
- [x] `package-lock.json` will no longer be apart of your repo
- [x] `yarn.lock` will be apart of your repo
- [x] Commands in repo's README use yarn, not npm
- [x] scripts in `package.json` use yarn over npm, when possible

## Notes
There is no _real_ way to default to one over the other. Just want your project to utilize yarn as much as possible over npm.

### PR DESCRIPTION
**Pull request by Jropp**
## Changes

1. The linter is added and set up in the scripts so you can either run `$ npm run linter` to just run the linter or `$ npm run start` to run the linter and start the server:

```
"scripts": {
    "build": "",
    "start": "npm run lint | python -m SimpleHTTPServer 1820",
    "help": "npm-scripts-help",
    "lint":  "ux-lint src/*.js"
  },
```

2. I've moved the files into an src folder and for the moment set up a test .js file. The code in there is inconsequential at the moment. I'll open an issue to move the javascript into that file

closes #24 (see cf74a1f)
closes #8
## A Problem

Running `$ npm run start` does run things appropriately, but it doesn't actually fire the linter. The server just ignores it. I'll do some more googling but I'm a little stuck on this point.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,10 +19,11 @@
   "bugs": {
     "url": "https://github.com/Jropp/jason-ropp-js-iobp/issues"
   },
-  "homepage": "https://github.com/Jropp/jason-ropp-js-iobp#readme",
```
Comment (kddrake) [inline package.json]: The readme was fine to have as the homepage, but honestly I'd just keep it as `https://github.com/Jropp/jason-ropp-js-iobp`

File: src/test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict';
```
Comment (kddrake) [inline src/test.js]: What's the purpose of this file? If it for nothing other than to ensure the linter is working, you should be able to lint the script in your user profile. Well, that's what's expected anyway. I'll updated the issue to make sure that's specified.

**Comment by Jropp** [comment]
@kddrake I think its working properly now. Plus the minor changes you recommended.

**Comment by Jropp** [comment]
@kddrake 

I found this at the bottom of eslint.hjson:

```
"settings": {
		"html/report-bad-indent": "error", // error if html/indent is bad
		"html/indent": "+2", // <script> indentation plus two spaces
		// lint JavaScript inside <script> tags with no `type` attribute, `type="module"`,
		// or `type="application/javascript"`
		"html/javascript-mime-types": "/^(module|application/javascript)$/",
	},
```
At a glance it looks like the RegExp is currently set properly.

**Comment by kddrake** [comment]
@Jropp try violating that bad indent error and see if the lint command fails. If it fails you're right, otherwise the JS isn't being linted.

**Comment by Jropp** [comment]
@kddrake tried it. It's not being linted.

**Comment by kddrake** [comment]
@Jropp so then have `ux-lint` watch for html files within src. I would assume that's all you need for it to do.

From what I understand after reading the readme, it defaults to only reading `*.js` files unless otherwise specified.

**Comment by Jropp** [comment]
@kddrake It's erroring out after compiling a list. Is it just not going to start the server until all of the listed issues are fixed?

**Comment by kddrake** [comment]
@Jropp yup, you're correct. should run the server like normal once you get you code up to snuff.

File: src/users-list.html [hunk 3]
```diff
@@ -52,37 +51,40 @@ <h2 class="name">Jason Ropp</h2>
...
-      function toggleDisplay(elem) {
-
-        // Bug Fix: The first instance on page load does not register the display block style, even when added explicitly in main.css.
-        if (elem.style.display === '') {
-          return elem.style.display = 'none';
-        } else {
-          (elem.style.display == 'block') ? elem.style.display = 'none' : elem.style.display = 'block';
-        }
+    function toggleDisplay(elem) {
+        // Bug Fix: The first instance on page load
```
Comment (kddrake) [inline src/users-list.html]: A slightly nicer way to format multiple line comment:
```
/**
 * Bug Fix: the first instance on page load
 * does not register the display block style,
 * even when added explicitly in main.css
 */
```
Doesn't need to be fix, just thought I'd share.

File: README.md [hunk 4]
```diff
@@ -6,16 +6,16 @@ This app allows you to create and edit user profiles, each of which have an ID,
 ## how to use
 
 ### NPM Scripts
-    To serve the app:
+    To serve the app (also runs ux-lint):
 
-        $ npm start 
+        $ yarn start 
 
     To build the app (Not currently functional):
```
Comment (kddrake) [inline README.md]: Since you took the script out, this should also be removed from the readme

File: src/new-user.html [hunk 5]
```diff
@@ -15,10 +15,10 @@
 <body>
     <header>
         <nav>
-            <a>User List</a>
-            <a>Create User</a>
-            <a>Edit User</a>
-            <a>User Profile</a>
+            <a href="something">User List</a>
```
Comment (kddrake) [inline src/new-user.html]: Use the hash `#` rather than `something` for null hrefs

File: src/user-profile.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    if (elem.style.display === '') {
+      elem.style.display = 'none';
+
+    } else {
+      elem.style.display = (elem.style.display === 'block') ? 'none' : 'block';
+    }
+  };
+  
+
+  let toggleUserDetails = function(event) { // eslint-disable-line
```
Comment (kddrake) [inline src/user-profile.html]: You don't want the linter to skip the line entirely, so specify which test you want disabled

**Comment by Jropp** [comment]
testing slack/github app

File: src/edit-user.html [hunk 7]
```diff
@@ -13,10 +13,10 @@
 <body>
     <header>
         <nav>
-            <a href="./src/users-list.html">User List</a>
-            <a href="./src/new-user.html">Create User</a>
-            <a href="./src/edit-user.html">Edit User</a>
-            <a href="./src/user-profile.html">User Profile</a>
+          <a href="http://localhost:1820/src/users-list.html">User List</a>
```
Comment (kddrake) [inline src/edit-user.html]: not quite. you just need `./*.html`. The `./` denotes "this directory", so `src/` is unnecessary. The new way doesn't navigate, either.

**Pull request merged**